test/IPCSocket.cpp

```diff
@@ -0,0 +1,138 @@
+/*
+	This file is part of cpp-ethereum.
+
+	cpp-ethereum is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	cpp-ethereum is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with cpp-ethereum.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/** @file IPCSocket.cpp
+ * @author Dimtiry Khokhlov <dimitry@ethdev.com>
+ * @date 2016
+ */
+
+#include <string>
+#include <stdio.h>
+#include "IPCSocket.h"
+using namespace std;
+
+//Just working example of the node configuration file
+std::string const c_genesisConfiguration = R"(
+{
+	"sealEngine": "NoProof",
+	"options": {
+		"authorities": [
+			"0xfa0c706a1410c8785baa7498325cf7461b325583",
+			"0x7766d151b2c63cb096f624daa091ccb27a2c693f"
+		]
+	},
+	"params": {
+		"accountStartNonce": "0x",
+		"maximumExtraDataSize": "0x1000000",
+		"blockReward": "0x",
+		"registrar": "",
+		"networkID" : "0x25",
+	"difficulty" : "0x13141"
+	},
+	"genesis": {
+		"author": "0x1000000000000000000000000000000000000000",
+		"timestamp": "0x00",
+		"parentHash": "0x0000000000000000000000000000000000000000000000000000000000000000",
+		"extraData": "0x",
+		"gasLimit": "0x1000000000000"
+	},
+	"accounts": {
+		"0000000000000000000000000000000000000001": { "wei": "1", "precompiled": { "name": "ecrecover", "linear": { "base": 3000, "word": 0 } } },
+		"0000000000000000000000000000000000000002": { "wei": "1", "precompiled": { "name": "sha256", "linear": { "base": 60, "word": 12 } } },
+		"0000000000000000000000000000000000000003": { "wei": "1", "precompiled": { "name": "ripemd160", "linear": { "base": 600, "word": 120 } } },
+		"0000000000000000000000000000000000000004": { "wei": "1", "precompiled": { "name": "identity", "linear": { "base": 15, "word": 3 } } }
+	},
+	"network": {
+		"nodes": [
+				 ]
+	}
+}
+)";
+
+IPCSocket::IPCSocket(string const& _path): m_address(_path)
+{
+	if (_path.length() > 108)
+		BOOST_FAIL("Error opening IPC: socket path is too long!");
+
+	struct sockaddr_un saun;
+	saun.sun_family = AF_UNIX;
+	strcpy(saun.sun_path, _path.c_str());
+
+	if ((m_socket = socket(AF_UNIX, SOCK_STREAM, 0)) < 0)
+		BOOST_FAIL("Error creating IPC socket object");
+
+	int len = sizeof(saun.sun_family) + strlen(saun.sun_path);
+
+	if (connect(m_socket, reinterpret_cast<const struct sockaddr *>(&saun), len) < 0)
+		BOOST_FAIL("Error connecting to IPC socket: " << _path);
+
+	m_fp = fdopen(m_socket, "r");
+}
+
+string IPCSocket::sendRequest(string const& _req)
+{
+	send(m_socket, _req.c_str(), _req.length(), 0);
```

Comment:
dunno. that was socket example. perhaps just using iostream can't write to the socket.